### ISSUE TITLE
Repo no longer maintained? Everything is way outdated

### ISSUE THREAD

**Issue by DerHerrGammler** [scullyio/scully#1660]
For me it seems that this repo is no longer active maintained and updated to the latest versions and features of angular.  
Do anyone plan to update the repo or is this repo droped and nobody care anymore?

I realy like the idea of this repo but when its not updated anymore it is probably not worth the try

**Comment by SanderElias** [comment]
@DerHerrGammler We are working on a new release. Everything is done except for the schematics. 
If you want to help DM me, and you are more then welcome to help. Once everything is in place we will update this repo again.

**Comment by jessycormier** [comment]
Sounds great @SanderElias! Is there a platform used to post updates that your community could use to see what's been going on?


-- edit

Noticed ScullyIO [Gitter channel](https://gitter.im/scullyio/community) is prob the best place :)

**Comment by fireflysemantics** [comment]
Awesome!!  Thanks for getting back to us!!

**Comment by Deliana90** [comment]
RIP @_@

**Comment by d-koppenhagen** [comment]
@SanderElias is there any update about the future of Scully? I mean will it be pushed and updated soon or is it more in maintenance mode and there is no plan to actively develop on it? I really liked scully but since a while it seems this repo won't be updated anymore at all. For example it won't work properly with Angular 16 or 17 since the peerDependencies they request doesn't match.

**Comment by SanderElias** [comment]
We have an updated `@beta` or `@next` version on NPM. We have done a lot of work that is not yet public.
But once the hydration story of angular came into play, things changed. Now with V17 even more things are set in motion.
I'm looking into ways where I can embrace all the work that is being done for `@angular@ssr` into the Scully workflow.
There are still a few missing parts, and we are looking into solutions. 
This might take a while (remember, all of use working on Scully are doing so in their spare time!)

Scully is currently in schrödinger mode. It might be a bit longer before it gets out of that, and depending what is possible with the current evolvements in Angular.

**Comment by djFooFoo** [comment]
Is there any plans on updating to angular v17?

Let me put it in a different way:
- I'm currently using Scully + TransferState service to pre-render an application, before it is put in a bucket and served by cloudfront. It's awesome, I love it.

- Angular 17 now comes with this SSR. Will it completely replace the role of Scully, or are there still many advantages of using Scully overall? I'm already getting compatibility warnings, but I'm still able to upgrade to a working version. Any updates coming in the near future to scully?

- Are there any contributions we could do, to make it at least compatible with the latest angular version etc?

**Comment by SanderElias** [comment]
@djFooFoo 
SSR is not new in V17. It predates Scully by a long time. Also, it is already compatible. The 3rthpary guess-parser isn't, so the automated route detection will only work when you are still using ngModules for all routing stuff, and don't do lazy loading. 

V17 comes with a form of route discovery, but this is still not accessible by Scully. I'm talking to the team to get access to integrate this and regain the automated route discovery.

The errors you get with npm install result from the package.json restrictions. There will be a new release that doesn't error, but while so many things are still up in the air, oh well, read my previous message in this thread.

There will be a next version that addresses everything, but it will take time. Everyone currently involved with Scully is doing this in their spare time. We are all very busy. 
A substantial sponsorship is the best contribution. 

I will try pushing out a new beta version in the upcoming weeks, to get at least rid of the npm errors.

**Comment by CesarD** [comment]
>Angular 17 now comes with this SSR. Will it completely replace the role of Scully, or are there still many advantages of using Scully overall? I'm already getting compatibility warnings, but I'm still able to upgrade to a working version. Any updates coming in the near future to scully?

I think this has not been addressed in the last comment and I'd like to understand the extent of the SSR stuff that now is baked into Angular 17... Is Scully still relevant in front of Angular 17 considering that we can also run pre-rendering natively on Angular?

**Comment by fireflysemantics** [comment]
Just tested out prerendering in Angular 17 and as far as I can tell it works the same way as in scully essentially.

https://fireflysemantics.medium.com/creating-an-angular-application-with-route-prerendering-39d6cd4df22a

The only thing that I noticed is that if we include routes in `routes.txt` that don't match route paths Angular still renders them as static routes.  I filed a feature request to fix this here:

https://github.com/angular/angular/issues/53522

Other than that it looks like Apples to Apples so far.

**Comment by CesarD** [comment]
What I've been able to test is that Angular's pre-rendering capabilities only work for the initial request to the app. At that point it will load the pre-rendered page and, afterwards, the remaining bundle of the SPA, so that's what it's going to keep loading afterwards, performing requests against whatever you're using as backend as it would normally do.
The difference that I'm noticing in all this is that, despite generating all static pages, this won't just use them all permanently, which means that it will still require hitting and pulling the data that populates your pages.
I understand Scully actually generates the static pages with all data already populated, am I right? But it still isn't fully compatible with Angular 17?
I'm attempting to create a blog site where I can pre-render the pages during a build process and then upload everything to a hosting, pretty much like Nuxt Content does in Nuxt (VueJs). Would this be possible with Scully?

**Comment by anuj9196** [comment]
Is this library still maintained? I can't see any update in the source code after 2022.

**Comment by mohitkanwar** [comment]
They are trying to change many things altogether and then getting stuck.
They are trying to achieve quality by providing 0 bugs, causing time delays..
My suggestion is to start releasing small, non-perfect releases.

(Also, it is easier to give suggestions when I am not the one building it. So we would never understand the complexities until we are part of the development process)

**Comment by hisham** [comment]
We recently updated to Angular 17 and the biggest thing we see now is the delay in Scully transitioning from the static site to the dynamic site. Angualr no longer exports the Zone Task Tracker, so scully defaults to just waiting 5 seconds before it transitions to the angular site. This affects our site performance/usability. See https://github.com/scullyio/scully/issues/1675

Any workarounds / fixes for this issue? It feels like as the v3 beta is developed, we should at least have hot fixes for the v2 version to keep it alive and performing ok.

**Comment by hisham** [comment]
It seems like with Angular 17.3, Angular is a lot better in transitioning from static pre-rendered site to dynamic SPA app. There is no flash in between. So things like flash prevention plugin are no longer needed. 

I also tested the angular prerender command in 17.3 and it works really well. It seems like Angular now supports the core part of Scully. Perhaps that makes the job of Scully team easier. Angular prerender command is great. We just need now a build pipeline that Scully provides to minify images, add SEO tags, convert markdown to blog html pages, etc...

**Comment by sanjay51** [comment]
Hi, is this repo still supported?

**Comment by jessycormier** [comment]
You don't really need this repo or tooling anymore since angular supports ssg.
